tests/test_util.py

```diff
@@ -74,3 +86,68 @@ def test_zipsafe():
     for zip_safe in (False, True):
       with make_egg(zipped=zipped, zip_safe=zip_safe) as dist:
         assert DistributionHelper.zipsafe(dist) is zip_safe
+
+
+@mock.patch('pex.util.resource_string', spec=pkg_resources.resource_string)
+@mock.patch('pex.util.resource_isdir', spec=pkg_resources.resource_isdir)
+@mock.patch('pex.util.resource_listdir', spec=pkg_resources.resource_listdir)
+def test_access_zipped_assets(mock_resource_listdir, mock_resource_isdir, mock_resource_string):
+  try:
+    import __builtin__
+    builtin_path = __builtin__.__name__
+  except ImportError:
+    # Python3
+    import builtins
+    builtin_path = builtins.__name__
+
+  mock_open = mock.mock_open()
+  with mock.patch('%s.open' % builtin_path, mock_open, create=True):
```

Comment:
not sure if this is available in our version but we should use it:
https://docs.python.org/3/library/unittest.mock-helpers.html#mock-open

if that's not available, '__builtins__' is consistent in python 2.x and python 3.x